proposed/2022/vulnerabilities-in-restore.md

```diff
@@ -0,0 +1,392 @@
+# Auditing projects for package vulnerabilities during restore
+
+- [Jon Douglas](https://github.com/JonDouglas/), [Nikolche Kolev](https://github.com/nkolev92), [Peter Yu](https://github.com/ryuyu)
+- Issue: [#8087](https://github.com/NuGet/Home/issues/8087)
+- Related: [#11549](https://github.com/NuGet/Home/pull/11549)
+
+## Summary
+
+Audit your dependencies when restoring your NuGet packages with security vulnerabilities reported to GitHub Advisory Database.

```

Comment:
The current proposal is that NuGet client gets the list of vulnerabilities from the NuGet package source. How the server implements the vulnerability information resource is up to the server implementation. They certainly could use osv.dev.

Looking at osv.dev, I don't see a way to download the entire database (unless I need to clone, build, and run the github repo?). So, from a NuGet client point of view, osv.dev doesn't fit well with our design.

This could be something added in the future if there's enough demand, but the current plan is that NuGet server implementations are responsible for providing the client with the relevant data.

Reply:
Thanks to clarify on that, maybe it's worth adding to the docs? 

FYI there are data dumps available:
https://github.com/google/osv.dev#data-dumps

Reply:
And more important: GH advisory db is a source of osv.dev! 😉

Reply:
I updated the spec to try to make this clearer.